### PR TITLE
add a new welcome banner to nushell

### PR DESCRIPTION
**Pull request by fdncred**
# Description

This PR adds a new welcome banner to nushell.
Screen Shot 2022-07-28 at 8 22 46 AM

The banner can be disabled with `show_banner: false` in the config.nu. It also respects the `use_ansi_coloring` setting if your terminal doesn't support colors,

# Tests

Make sure you've done the following:

- [ ] Add tests that cover your changes, either in the command examples, the crate/tests folder, or in the /tests folder.
- [ ] Try to think about corner cases and various ways how your changes could break. Cover them with tests.
- [ ] If adding tests is not possible, please document in the PR body a minimal example with steps on how to reproduce so one can verify your change works.

Make sure you've run and fixed any issues with these commands:

- [x] `cargo fmt --all -- --check` to check standard code formatting (`cargo fmt --all` applies these changes)
- [x] `cargo clippy --workspace --features=extra -- -D warnings -D clippy::unwrap_used -A clippy::needless_collect` to check that you're using the standard code style
- [x] `cargo test --workspace --features=extra` to check that all the tests pass

### CODE REVIEW

**Comment by jntrnr** [comment]
I wonder, should we tell them about `config nu` to edit the file, so they can more easily edit it?

**Comment by fdncred** [comment]
@jntrnr How about this?
- added url colors
- changed text based on your suggestion
- changed disable text part to a different color
Screen Shot 2022-07-28 at 6 44 36 PM

**Comment by onthebridgetonowhere** [comment]
Great idea and execution, I dig it!! 
It will also (hopefully) help attract more contributors to the project and to join the Discord channel.

**Comment by jntrnr** [comment]
@fdncred looking good - I think the last touch would be setting `config nu` in a different highlight than the rest to make it clearer it's a command name with two parts. But yeah looks great

**Comment by fdncred** [comment]
This one makes `config nu` stand out a little using dim cyan.
Screen Shot 2022-07-29 at 6 50 09 AM

File: crates/nu-cli/src/repl.rs [hunk 1]
```diff
@@ -460,6 +481,121 @@ pub fn evaluate_repl(
...
+    let banner = format!(
+        r#"{}     __  ,
+{} .--()°'.' {}Welcome to {}Nushell{}, 
+{}'|, . ,'   {}based on the {}nu{} language,
+{} !_-(_\    {}where all data is structured!
+
+Please join our {}Discord{} community at {}https://discord.gg/NtAbbGn{}
+Our {}GitHub{} repository is at {}https://github.com/nushell/nushell{}
+Our {}Documentation{} is located at {}http://nushell.sh{}
+{}Tweet{} us a {}@nu_shell{}
```
Comment (sholderbach) [inline crates/nu-cli/src/repl.rs:504]: Maybe either
```suggestion
{}Tweet{} us {}@nu_shell{}
```

or `Tweet us at ...`

Comment (fdncred) [reply crates/nu-cli/src/repl.rs:504]: good catch. i'll update it.

**Comment by fdncred** [comment]
image

**Comment by sholderbach** [comment]
Are those `xterm-256` colors or base 4bit/16 colors?

**Comment by fdncred** [comment]
@sholderbach These are traditional colors, nothing fancy.
1. 32 - green
2. 35 - purple
3. 1;32 - green bold
4. 36 - cyan
5. 1;36 - cyan bold
6. 2;37 - dim white
7. 2;36 - dim cyan

**Pull request merged**